src/algebra/order/rearrangement.lean

```diff
@@ -12,12 +12,21 @@ import tactic.abel
 /-!
 # Rearrangement inequality
 
-This file proves the rearrangement inequality.
+This file proves the rearrangement inequality and deduces the conditions for equality and strict
+inequality
 
 The rearrangement inequality tells you that for two functions `f g : ι → α`, the sum
 `∑ i, f i * g (σ i)` is maximized over all `σ : perm ι` when `g ∘ σ` monovaries with `f` and
 minimized when `g ∘ σ` antivaries with `f`.
 
+The inequality also tels you that if `∑ i, f i * g (σ i) = ∑ i, f i * g i` if and only `g ∘ σ`
+monovaries with `g` when `f` monovaries with `f`. The above equality also holds if and only if
+`g ∘ σ` antivaries with `f` when `g` antivaries with `f`.
```

Comment:
It sounds as if "`g ∘ σ` antivaries with `f` when `g` antivaries with `f`" if and only if "`g ∘ σ` monovaries with `f` when `g` monovaries with `f`". Is that right?

Reply:
That was not the statement I was trying to spell out. Fixed in the new commit now